src/sentry/api/endpoints/organization_metrics_meta.py

```diff
@@ -0,0 +1,113 @@
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.bases import NoProjects, OrganizationEventsEndpointBase
+from sentry.search.events.fields import get_function_alias
+from sentry.snuba import metrics_performance
+
+COUNT_UNPARAM = "count_unparameterized_transactions()"
+COUNT_HAS_TXN = "count_has_transaction_name()"
+COUNT_NULL = "count_null_transactions()"
+
+
+class OrganizationMetricsCompatibility(OrganizationEventsEndpointBase):
+    """Metrics data can contain less than great data like null or unparameterized transactions
+
+    This endpoint will return projects that have dynamic sampling turned on, and another list of "compatible projects"
+    which are the projects which don't have null transactions and have at least 1 transaction with a valid name
+    """
+
+    private = True
+
+    def get(self, request: Request, organization) -> Response:
+        data = {
+            "compatible_projects": [],
+            "dynamic_sampling_projects": [],
+        }
+        try:
+            # This will be used on the perf homepage and contains preset queries, allow global views
+            params = self.get_snuba_params(request, organization, check_global_views=False)
+        except NoProjects:
+            return Response(data)
+        data["compatible_projects"] = params["project_id"]
+        for project in params["project_objects"]:
+            dynamic_sampling = project.get_option("sentry:dynamic_sampling")
+            if dynamic_sampling is not None:
+                data["dynamic_sampling_projects"].append(project.id)
+                if len(data["dynamic_sampling_projects"]) > 50:
+                    break
+
+        # None of the projects had DS rules, nothing is compat the sum & compat projects list is useless
+        if len(data["dynamic_sampling_projects"]) == 0:
+            return Response(data)
+        data["dynamic_sampling_projects"].sort()
+
+        # Save ourselves some work, only query the projects that have DS rules
+        params["project_id"] = data["dynamic_sampling_projects"]
+
+        with self.handle_query_errors():
+            count_has_txn = "count_has_transaction_name()"
+            count_null = "count_null_transactions()"
+            compatible_results = metrics_performance.query(
+                selected_columns=[
+                    "project.id",
+                    count_null,
+                    count_has_txn,
+                ],
+                params=params,
+                query=f"{count_null}:0 AND {count_has_txn}:>0",
+                referrer="api.organization-events-metrics-compatibility.compatible",
+                functions_acl=["count_null_transactions", "count_has_transaction_name"],
+                use_aggregate_conditions=True,
+            )
+            data["compatible_projects"] = sorted(
+                row["project.id"] for row in compatible_results["data"]
+            )
```

Comment:
Can you add `incompatible_projects` so we can check the type of incompatibility? Or do we want to wait until the sdk's are updated?